openwisp_monitoring/device/api/serializers.py

```diff
@@ -20,6 +25,72 @@ class Meta:
         model = DeviceMonitoring
 
 
+class WifiClientSerializer(FilterByOrganizationManaged, ValidatedModelSerializer):
+    class Meta:
+        fields = [
+            "mac_address",
+            "vendor",
+            "ht",
+            "vht",
+            "wmm",
+            "wds",
+            "wps",
+            "modified",
+            "created",
+        ]
+        model = WifiClient
+        read_only_fields = (
+            'created',
+            'modified',
+        )
+
+
+class WifiSessionCreateUpdateSerializer(
+    FilterByOrganizationManaged, ValidatedModelSerializer
+):
+    # When a relationship or ChoiceField has too many items,
+    # rendering the widget containing all the options can become very slow,
+    # and cause the browsable API rendering to perform poorly, Changed select field to input.
+    device = serializers.PrimaryKeyRelatedField(
+        queryset=Device.objects.all(), style={'base_template': 'input.html'}
+    )
+    wifi_client = serializers.PrimaryKeyRelatedField(
+        queryset=WifiClient.objects.all(), style={'base_template': 'input.html'}
+    )
+
+    class Meta:
+        model = WifiSession
+        fields = ['device', 'wifi_client', 'ssid', 'interface_name']
+
+
+class WifiSessionReadSerializer(FilterByOrganizationManaged, ValidatedModelSerializer):
```

Comment:
We should explore the possibility of using just one serializer. What problems did you face that made you split up the serializer?

Reply:
On creating a WifiSession, we only get following fields in the response. 

![Screenshot from 2022-06-07 16-45-35](https://user-images.githubusercontent.com/32094433/172366624-dabdadce-df16-424a-93c5-24d33c58db01.png)

Reply:
Let's wait for @nemesisdesign opinion before proceeding with this change.

Reply:
I asked to have nested serializer because I think it would work well here, see https://github.com/openwisp/openwisp-monitoring/pull/382#discussion_r885068006.